readme-docs/KUBERNETES.md

```diff
@@ -375,6 +375,84 @@ extraEnv:
     value: '26379'
 ```
 
+
+### How to use an external PostgreSQL DB with Defectdojo
+
+#### Step 1: Create a Namespace for DefectDojo
+
+To begin, create a dedicated namespace for DefectDojo to isolate its resources:
+`kubectl create ns defectdojo`
+#### Step 2: Create a Secret for PostgreSQL Credentials
+
+Set up a Kubernetes Secret to securely store the PostgreSQL user password and database connection URL, which are essential for establishing a secure connection between DefectDojo and your PostgreSQL instance. Apply the secret using the following command: `kubectl apply -f secret.yaml -n defectdojo`. This secret will be referenced within the `extraEnv` section of the DefectDojo Helm values file.
+
+Sample secret template (replace the placeholders with your PostgreSQL credentials):
+```YAML
+apiversion: v1
+kind: Secret
+metadata:  
+	name: defectdojo-postgresql-specific  
+type: Opaque
+stringData:  # I chose stringData for better visualization of the credentials for debugging
+postgresql-password: <user-password>
+db-url: psql://<username>:<password>@<hostname>:<port>/<database-name> 
+```
+
+#### Step 2.5: Install PostgreSQL (Optional)
+
+If you need to simulate a PostgreSQL database external to DefectDojo, you can install PostgreSQL using the following Helm command:
+
+```helm
+helm repo add bitnami https://charts.bitnami.com/bitnami 
+helm repo update
+helm install defectdojo-postgresql bitnami/postgresql -n defectdojo -f postgresql/values.yaml
+```
+
+
+Sample `values.yaml` file for PostgreSQL configuration:
+
+```YAML 
+auth:
+  username: defectdojo
+  password: <user-password>
+  postgresPassword: <admin-password>
+  database: defectdojo
+  primary:
+    persistence:
+    size: 10Gi
+```
+
+#### Step 3: Modify DefectDojo helm values
+
+Before installing the DefectDojo Helm chart, it's important to customize the `values.yaml` file. Key areas to modify include specifying the PostgreSQL connection details & the extraEnv block:
+
+```yaml
+database: postgresql
+postgresql:
+  postgresServer: "defectdojo-postgresql" # point to the hostname of your postgresql server
```

Comment:
It doesn't work without that value, for some reason it breaks the cm in the defectdojo chart:

![image](https://github.com/user-attachments/assets/b9cfef13-7b7b-411e-a230-9dfef455ba43)

I also noticed something similar with the custom secret I created for external postgres credentials. Even though I've already specified the dd_url which contains the username,password etc. The helm deployment fails unless a `postgres-password` key is present.

Reply:
It might be a good idea to adjust the original HELM chart so that useless information is skipped.

Reply:
@kiblik 
I looked into this some more and discovered that the postgresServer entry resolves to DD_DATABASE_HOST.  This is used in the initializer job to check if the db is up. 
![image](https://github.com/user-attachments/assets/77ff011d-00d2-46af-8e76-5d8c05db7cf0)

A workaround would be to add DD_DATABASE_HOST to the extraEnv section but I'm not sure that gives any additional value.

Reply:
@kiblik any update on this?

Reply:
Oh no. You are right. I forgot about this one.
There would be still an option to add the condition to HELM: "If DD_DATABASE_URL is defined, parse it to DD_DATABASE_HOST, DD_DATABASE_PORT, ..." but it is a bit dirty.

Plus in that case, it would be fair to adopt all other places like https://github.com/search?q=repo%3ADefectDojo%2Fdjango-DefectDojo+path%3A%2F%5Edocker%5C-compose*.%2F+DD_DATABASE_HOST&type=code

Maybe the requirement "`DD_DATABASE_HOST` and `DD_DATABASE_PORT` must be defined" is the easier way?

Reply:
@kiblik 
Yes I agree, trying to parse it might be a bit dirty. I believe requiring `postgresql.postgresServer` in the values file and `postgresql-password & db-url` in the custom postgres secret is the way to go. I documented all the steps in this pr. I've already deployed DD using this same config and its working great so far.

Reply:
@kiblik 
On a sidenote, are you aware if I can use valkey as a drop-in replacement for redis as the message broker?